### ISSUE TITLE
cmd/link: impossible r.Sym.Cfunc == 1 condition in internal/ld/lib.go

### ISSUE THREAD

**Issue by mdempsky** [golang/go#13257]
cmd/link has this test in a block of code labeled "Mark every Go-called C function with cfunc=2, recursively.":

```
internal/ld/lib.go:2096:                                                if r.Sym.Cfunc == 1 {
```

As far as I can tell, the only assignments to cmd/link/internal/ld.LSym's Cfunc field are:

```
internal/ld/lib.go:2098:                                                        r.Sym.Cfunc = 2
internal/ld/objfile.go:268:             s.Cfunc = uint8(v & 2)
```

neither of which can set it to 1.

(There's a "ptxt.From.Sym.Cfunc = 1" assignment in compile/internal/gc/pgen.go, but that's referring to the Cfunc field of an cmd/internal/obj.LSym.)

FWIW, I tracked this check back to https://golang.org/cl/135170044 (0bb14d74f1550dca916251ddb6e1703f95be7325), and it appears to have been impossible even then (though less obvious since cmd/cc and cmd/ld used the same LSym type).

CC @rsc @randall77

**Comment by mwhudson** [comment]
I think this is some of the code I was talking about here: https://groups.google.com/d/msg/golang-dev/KK5olmcQ6U4/RdnUHhkVeA8J

**Comment by mdempsky** [comment]
@mwhudson Yep, though whereas you were pointing out the code was unneeded after the c2go conversion (until rsc later repurposed it for go:systemstack), I'm pointing out that "cfunc == 1" is (and has always been) impossible within the linker.

Because of an asymmetry between the code responsible for writing cfunc into a Go object file and reading it back in, setting cfunc==1 in the compiler results in setting cfunc==2 in the linker, and there's no other assignments of cfunc=1 within the linker.

**Comment by mwhudson** [comment]
Ah OK :-)

**Comment by rsc** [comment]
Yes, func checkgo can be deleted.

**Comment by ALTree** [comment]
Fixed by 1012892f1e6329d5581ad82a20ababa5548d8dc2.
